x.py

```diff
@@ -106,7 +106,7 @@ def build(dir: str, jobs: Optional[int], ghproxy: bool, ninja: bool, unittest: b
 
     makedirs(dir, exist_ok=True)
 
-    cmake_options = ["-DCMAKE_BUILD_TYPE=RelWithDebInfo"]
+    cmake_options = ["-DCMAKE_BUILD_TYPE=RelWithDebInfo", "-DPORTABLE=0"]
```

Comment:
Why do we need to add `PORTABLE=0` here? I saw the default value for the PORTABLE option is 0 now.

Reply:
I have no idea, why, but build are Ok only when we manual add -DPORTABLE=0. In all other cases:
cc1plus: error: bad value ‘OFF’ for ‘-march=’ switch

Reply:
As I see a cmake docs: "If no initial <value> is provided, boolean OFF is the default value", so we need a manual add 0 as value

Reply:
It should be broken by this PR: https://github.com/facebook/rocksdb/pull/11419. But I think we can remove `"-DPORTABLE=0"` since the default value has been corrected to 0 in the previous line: https://github.com/apache/kvrocks/pull/1516/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR32

Reply:
If we add 0 or 1 as value (in option) cmake rewrite it into ON/OFF and only manual setup are valid

Reply:
> It should be broken by this PR: [facebook/rocksdb#11419](https://github.com/facebook/rocksdb/pull/11419). But I think we can remove `"-DPORTABLE=0"` since the default value has been corrected to 0 in the previous line: https://github.com/apache/kvrocks/pull/1516/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR32

No, default value, if we set 0 or 1, cmake rewrite into own default ON/OFF

Reply:
oh, I got your point. Would it be better to remove the PORTABLE option in CMakeList.txt? since the rocksdb will set the default value to 0 if the PORTABLE was not passed. cc @PragmaTwice

Reply:
And this way not a good point - now we can manual add cpu arch for better optimization (much better as default) and if completely remove options, there no way to do this anymore

Reply:
What I mean is only to remove the option from the Kvrocks CMakeList.txt to prevent automatically rewriting it into ON/OFF, users still can pass the PORTABLE via the CMake arguments. Did I miss anything?

Reply:
Yes, but we need manual setup always, because in rocksdb.cmake we use it.

So, only one right way in now - always manual set PORTABLE as a Cmake flag on build

Reply:
Yes, i send issue for rocksdb team

Reply:
@aleksraiden Is it public and we can cross-post the link here?